src/Support/FileNamer/FileNamer.php

```diff
@@ -9,9 +9,9 @@ abstract class FileNamer
 {
     public function originalFileName(string $fileName): string
     {
-        $name = pathinfo($fileName, PATHINFO_FILENAME);
+        $ext_length = strlen(pathinfo($fileName, PATHINFO_EXTENSION));
 
-        return preg_replace("/{$name}\..*/", $name, $fileName);
+        return substr($fileName, 0, strlen($fileName) - ($ext_length ? $ext_length + 1 : 0));
```

Comment:
To me `strlen($fileName) - ($ext_length ? $ext_length + 1 : 0` is not very readable. Maybe put it the result a new variable with a meaningful name?